### ISSUE TITLE
Weather screen latest issues

### ISSUE THREAD

**Issue by Nine-Tailed** [osmandapp/OsmAnd#18252]
### Description

On the weather screen there are two buttons labeled "today".
What is interesting is that they are available only at certain hours of the day, for example, I have now 23.50 Moscow time.
I would also like to note that the scroll bar of the clock fits on the numbers, in previous versions so was not.
And the day of the week in the buttons is more beautiful from formatting in the center.
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/12549744/fbd1dea0-d65c-48b8-b386-12eb86a8a09d"  width="300" height="auto"/> </td>
  </tr>
</table>

### Steps to reproduce

Turn on the weather screen at approximately 23:50.

### Actual result

Described above.

### Expected result

Result - one button today, the clock scale should not fit on the numbers. Day of the week centered.

### Your Environment (required)

```
OsmAnd Version:OsmAnd~ 4.6.0#815m, Дата выпуска: 2023-09-28
Android/iOS version: 13
Device model: росо ф3
Crash-Logs: ?
```

**Comment by DmitryAlexei** [comment]
OsmAnd~ 4.6.0#825m, released: 2023-10-01

**Steps to reproduce:**
- set the system time for the last half hour of the day (between 23.30 and 23.59) (on Samsung device go to Android settings > General management > date and time > disable automatic date and time and enter needed values)
- open OsmAnd app > Weather tab

**Actual result:**
- see screenshot

<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/fe208c99-1eb1-48ab-94a7-37559a1be984"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by vshcherb** [comment]
To do:
- [ ] Fix UI issue with numbers
- [ ] Fix mentionned Issue with 23.30 - 00.00
- [ ]  Check RTL has no issue with progress bar
